ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -587,7 +603,7 @@ private IndexValue findKey(StoreKey key, FileSpan fileSpan, EnumSet<IndexEntryTy
         logger.trace("Searching for {} in index with filespan ranging from {} to {}", key, fileSpan.getStartOffset(),
             fileSpan.getEndOffset());
         segmentsMapToSearch = indexSegments.subMap(indexSegments.floorKey(fileSpan.getStartOffset()), true,
-            indexSegments.floorKey(fileSpan.getEndOffset()), true).descendingMap();
+            indexSegments.lowerKey(fileSpan.getEndOffset()), true).descendingMap();
```

Comment:
End offset in FileSpan is exclusive so we don't have to include the end offset. Assuming we have several index segments and the map of indexSegments look like this (it's offset to index segment).
{ 
    0       :index segment1
    1000 :index segment2
    2000 :index segment3
    3000 :index segment4
}
When the filespan is [500, 2000), then we should only search on index segment1 and index segment2. index segment 3's start offset is 2000, but 2000 in filespan is exclusive, we don't need it.